### ISSUE TITLE
Single threaded diarization

### ISSUE THREAD

**Issue by alexandrefelt-droid** [homelab-00/TranscriptionSuite#173]
Improvement Idea: would it be possible to have Diarization running on more than 1 core?

**Comment by homelab-00** [comment]
Hmm, diarization is handled by WhisperX so I'm guessing that's a limitation on their part.
I will look into it though.

**Comment by homelab-00** [comment]
Looked into this - good news is it's mostly working as intended, but the answer depends on whether you're running diarization on a GPU or on CPU.

First, a small correction to my earlier guess: diarization is actually **PyAnnote** under the hood (WhisperX just wraps it), and the suite imposes **no single-core limit** anywhere in its code. So what "1 core" means depends on your setup:

**If you're on a GPU (the default):** the heavy work - the segmentation and speaker-embedding neural nets - runs *on the GPU*, which is massively parallel. Only PyAnnote's final speaker-clustering step runs on the CPU, and that step is single-threaded *by design in PyAnnote*. So seeing one busy CPU core in `htop` during diarization is completely normal and **not** a bottleneck - adding CPU cores wouldn't speed it up, because the real work is already parallel on the GPU.

**If you're on CPU-only:** PyTorch already spreads the neural phases across **all** your physical cores by default (nothing caps it). Again, only the clustering step is inherently single-threaded.

Could you check which device it's actually using? In the server logs you'll see a line like:

```
Diarization pipeline moved to device: cuda   (or mps / cpu)
```

and if it says `cuda`, run `nvidia-smi` during a job to confirm the GPU is active.

A couple of things I looked at and ruled out:
- **Raising CPU thread counts** (`OMP_NUM_THREADS` etc.) won't help - the cores are already saturated on the neural phases, and it can even slow things down.
- **Splitting the audio into chunks to diarize in parallel** sounds tempting but breaks speaker consistency: PyAnnote does one *global* clustering pass over the whole file, so per-chunk diarization gives the same person different speaker labels across chunks. Not worth the accuracy hit.

If diarization feels slow and you're on CPU, the single biggest win is running it on a GPU (roughly 10–30× faster) - that's already supported automatically when CUDA is available. Let me know what your device line says and we can go from there.
